### ISSUE TITLE
Create certain ANNs for classifying UrbanSound8K 

### ISSUE THREAD

**Issue by Shubhabrata08** [Shubhabrata08/AudioClassificationTFLite#3]
### Details:
Create ANNs for classifying UrbanSound8K. MFCCs of the audio samples are present in the linked dataset below and are to be directly used in the classification input as features.
Experiment on the ANNs, by modification of layers, addition/removal of layers, etc.
Feel free to try out ML algos on the features as well.
Do document the results in a text file with necessary details.
### Dataset:
[UrbanSound8KMFCCs](https://drive.google.com/file/d/1wJD1XCkFaT1TaszZBnFWJeCawW0EO-0D/view?usp=sharing)

### Tools and Libraries recommended:
Google Colab and Tensorflow.

**Comment by ShubhamJain357** [comment]
### *1. Completed Machine Learning and Deep Learning course By Krish Naik:-* 

- **_Learnt Machine Learning (Linear Regression, Logistic Regression, KMeans Clustering, KNN, Lasso and Ridge Regression, Decision Tree) and Ensemble Learning techniques ( Bagging, Boosting, Random Forest Classifier), [Machine Learning By Krish Naik](https://youtube.com/playlist?list=PLTDARY42LDV7WGmlzZtY-w9pemyPrKNUZ&feature=shared)_**
- **_Learnt Deep Learning( ANN and CNN classification techniques)(5 days deep learning course by Krish Naik), [5 days Deep Learning By Krish Naik](https://youtube.com/playlist?list=PLZoTAELRMXVPiyueAqA_eQnsycC_DSBns&feature=shared)_**

**Comment by ShubhamJain357** [comment]
### **Creating ANN for classifying UrbanSound8K**

**1. Reading h5py File**
```
import h5py

with h5py.File('/content/UrbanSound8kMFCC.h5py', 'r') as hdf_file:
    # List all the datasets in the HDF5 file/UrbanSound8kMFCC.h5py
    print("Datasets in the HDF5 file:", list(hdf_file.keys()))

    # Access the 'X' dataset
    X_dataset = hdf_file['X']

    # Access the 'y' dataset
    y_dataset = hdf_file['y']

    # Now, you can work with the datasets as NumPy arrays
    X_data = X_dataset[:]
    y_data = y_dataset[:]

    print("Shape of the 'X' dataset:", X_data.shape)
    print("Shape of the 'y' dataset:", y_data.shape)
```

**2.Creating an ANN Model**
```
import tensorflow as tf
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X_data, y_data, test_size=0.2, random_state=42)

# Normalize the input features
X_train = tf.keras.utils.normalize(X_train, axis=1)
X_test = tf.keras.utils.normalize(X_test, axis=1)

# Define the model
model = tf.keras.models.Sequential([
    tf.keras.layers.Input(shape=X_train.shape[1:]),
    tf.keras.layers.Dense(128, activation='relu'),
    tf.keras.layers.Dropout(0.2),
    tf.keras.layers.Dense(64, activation='relu'),
    tf.keras.layers.Dropout(0.2),
    tf.keras.layers.Dense(10, activation='softmax')  # Assuming 10 classes
])
```
**3. Training the model and Evaluating Accuracy**
```
# Compile the model
model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

# Train the model
history = model.fit(X_train, y_train, epochs=50, batch_size=32, validation_split=0.2)

# Evaluate the model on the test data
test_loss, test_accuracy = model.evaluate(X_test, y_test)
print("Test accuracy:", test_accuracy)
```
**4. Model Summary**
![image](https://github.com/Shubhabrata08/AudioClassificationTFLite/assets/73420744/331b5ccf-74b7-44a4-b12d-4b301e5a30f8)

**5. Accuracy and Loss Plot**
```
import matplotlib.pyplot as plt

# Plot training and validation accuracy curves
plt.plot(history.history['accuracy'], label='Training Accuracy')
plt.plot(history.history['val_accuracy'], label='Validation Accuracy')
plt.title('Training and Validation Accuracy')
plt.xlabel('Epoch')
plt.ylabel('Accuracy')
plt.legend()
plt.show()

# Plot training and validation loss curves
plt.plot(history.history['loss'], label='Training Loss')
plt.plot(history.history['val_loss'], label='Validation Loss')
plt.title('Training and Validation Loss')
plt.xlabel('Epoch')
plt.ylabel('Loss')
plt.legend()
plt.show()
```

**Output:**
![image](https://github.com/Shubhabrata08/AudioClassificationTFLite/assets/73420744/89dc54b4-ba99-423a-9e0f-e5e740860ab8)
![image](https://github.com/Shubhabrata08/AudioClassificationTFLite/assets/73420744/dc088cd3-888e-4932-9f6c-82933cb83194)
